pkg/network/netlink/conntracker.go

```diff
@@ -115,70 +94,40 @@ func NewConntracker(procRoot string, maxStateSize int, ignoreENOBUFS bool) (Conn
 	}
 }
 
-func newConntrackerOnce(procRoot string, maxStateSize int, ignoreENOBUFS bool) (Conntracker, error) {
-	netns := getGlobalNetNSFD(procRoot)
-	logger := getLogger()
-
-	nfct, err := createNetlinkSocket("register", netns, logger, ignoreENOBUFS)
+func newConntrackerOnce(procRoot string, maxStateSize, targetRateLimit int) (Conntracker, error) {
+	consumer, err := NewConsumer(procRoot, targetRateLimit)
 	if err != nil {
 		return nil, err
 	}
 
 	ctr := &realConntracker{
-		nfct:                   nfct,
-		compactTicker:          time.NewTicker(compactInterval),
-		state:                  make(map[connKey]*connValue),
-		maxStateSize:           maxStateSize,
-		exceededSizeLogLimit:   util.NewLogLimit(10, time.Minute*10),
-		staleConntrackLogLimit: util.NewLogLimit(1, time.Minute*10),
+		consumer:             consumer,
+		compactTicker:        time.NewTicker(compactInterval),
+		state:                make(map[connKey]*connValue),
+		maxStateSize:         maxStateSize,
+		exceededSizeLogLimit: util.NewLogLimit(10, time.Minute*10),
 	}
 
-	// seed the state
-	sessions, err := nfct.Dump(ct.Conntrack, ct.IPv4)
-	if err != nil {
-		return nil, err
-	}
-	ctr.loadInitialState(sessions)
-	log.Debugf("seeded IPv4 state")
-
-	sessions, err = nfct.Dump(ct.Conntrack, ct.IPv6)
-	if err != nil {
-		// this is not fatal because we've already seeded with IPv4
-		log.Errorf("Failed to dump IPv6")
-	}
-	ctr.loadInitialState(sessions)
-	log.Debugf("seeded IPv6 state")
-
-	go ctr.run()
-
-	nfct.Register(context.Background(), ct.Conntrack, ct.NetlinkCtNew, ctr.register)
-	log.Debugf("initialized register hook")
-
-	log.Infof("initialized conntrack with ignoreENOBUFS=%v", ignoreENOBUFS)
-
+	ctr.loadInitialState(consumer.DumpTable(unix.AF_INET))
```

Comment:
I don't really know this piece of the code base but it seems we had code path for for IPv4 & IPv6 before and now there is only a single call for IPv4, I'm just wondering if this is on purpose.

Reply:
Nice catch 🙇 I'll open a follow up PR adding the IPv6 dump/test-cases